submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionRequestDocDataRepository.kt

```diff
@@ -44,8 +47,17 @@ class SubmissionRequestDocDataRepository(
     private val extSerializationService: ExtSerializationService,
     private val submissionRequestRepository: SubmissionRequestRepository,
 ) : SubmissionRequestRepository by submissionRequestRepository {
-    fun saveRequest(submissionRequest: DocSubmissionRequest): DocSubmissionRequest =
-        submissionRequestRepository.save(submissionRequest)
+    fun saveRequest(request: DocSubmissionRequest): Pair<DocSubmissionRequest, Boolean> {
+        val result = mongoTemplate.upsert(
+            Query(where(RQT_ACC_NO).`is`(request.accNo).andOperator(where(RQT_STATUS).ne(PROCESSED))),
+            request.asSetOnInsert(),
+            DocSubmissionRequest::class.java
+        )
+        return when (val insertedId = result.upsertedId) {
+            null -> submissionRequestRepository.getByAccNoAndStatusIn(request.accNo, PROCESSING) to false
```

Comment:
This piece will always return the latest processing request so we could just return that one regardless